src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -455,7 +487,13 @@ void ReplicatedMergeTreeSink::commitPart(
                     else
                         quorum_path = storage.zookeeper_path + "/quorum/status";
 
-                    waitForQuorum(zookeeper, existing_part_name, quorum_path, quorum_info.is_active_node_value, replicas_num);
+                    if (!retries_ctl.call(
+                            [&]()
+                            {
+                                waitForQuorum(
+                                    zookeeper, existing_part_name, quorum_path, quorum_info.is_active_node_value, replicas_num);
+                            }))
+                        return;
```

Comment:
No, in case of exception in `waitForQuorum`, the exception is caught in `call()`, set corresponding error in retries control structure and `call()` returns false. Then execution will return from lambda to retry loop (see `retryLoop()`). There we'll see that there was an error, check if we can retry one more time, if so, then will retry